### DISCUSSION TITLE
Federated Module is undefined / ScriptExternalLoadError: Loading script failed.

### DISCUSSION THREAD

**Discussion by knightswatch3** [module-federation/module-federation-examples#3758] [Q&A]
Here is the [link](https://github.com/knightswatch3/federated-module-isse/tree/main) to the reproducible repo for the issue (a). Looking forward for some interactive discussion to understand and fix this issue. 

**Issue :**

   a. Loaded remote module from app1 is undefined. I am unable to use any of the store methods app1 is exporting via remote module.
   b. I have a similar issue with another app (not in the repo currently ) which exposes a react component called <Auth/> which when I import and use in the index.tsx of app2 I run into

   ```
      Error.tsx:45 Error: ScriptExternalLoadError: Loading script failed.
      (missing: http://localhost:3002/remoteEntry2.js)
      while loading "./src" from webpack/container/reference/authremote
      remote authremote/Auth 6 bytes (remote) 6 bytes (share-init) [built] [code generated]
      external "authremote@http://localhost:5002/remoteEntry2.js" 42 bytes [built] [code generated]
   ```

   I believe these two are inter-related and solving the first(8.a) will result in a fix for the other. 

# federated-module-issue

How to start ? 

1. Clone the repo.
2. Ensure you have pnpm (8.14.1) and node(v20.11.1).
3. cd into app1 and run pnpm run-app1
4. cd into app2 and run pnpm run-app2
5. This will spin up both the apps.

6. App1 Serves the remote module on 5001 as configured:

![image](https://github.com/knightswatch3/federated-module-isse/assets/9092073/481c6567-0e07-4b8a-bc2d-b2a9fbe6c90b)

7. App2 Serves the html for the react code composed.
   ![image](https://github.com/knightswatch3/federated-module-isse/assets/9092073/d5429369-c5ee-4444-9976-6bb57253df3d)

But app1 finds the imported federated module as undefined. Which is confusing. 

If I export a remote module which has a React components from *.tsx file I experience this issue:

![image](https://github.com/module-federation/module-federation-examples/assets/9092073/a5940ef4-d43a-43a8-97d2-65abab2e69ae)

What have I tried ? (and not worked)

1. Ensuring shared dependencies are valid. 
2. Ensure the network port doesn't issue any errors for remote module imports. 
3. Ensure that the remote module is actually serving the remoteEntry in the mentioned port(webpack config)
4. Passing library => var on the hosting side of the module to be available through the global scope of the app.
5. Passing the module as a promise from the remote object.

**Comment by ScriptedAlchemy** [comment]
Use v2 not the original webpack one.
